CI: Rustツールチェーンを1.56.1へアップグレードする（2021エディション対応）

2021エディションの章を追加したところ、CIで `mdbook test` が失敗するようになってしまった。現在CIで使用しているRustツールチェーンが2021エディションに対応していないのが原因。1.56.1以降にアップグレードする必要がある。

```console
#!/bin/bash -eo pipefail
mdbook test
...

--- stdout

running 1 test
test /tmp/mdbook-wjTpld/rust-2021/or-patterns-macro-rules.md - Or_patterns_in_macro_rules::Migration (line 54) ... FAILED

failures:

---- /tmp/mdbook-wjTpld/rust-2021/or-patterns-macro-rules.md - Or_patterns_in_macro_rules::Migration (line 54) stdout ----
error: invalid fragment specifier `pat_param`
  --> /tmp/mdbook-wjTpld/rust-2021/or-patterns-macro-rules.md:68:3
   |
16 |     ($x:pat_param | $y:pat) => { // <- this line is different
   |      ^^^^^^^^^^^^
   |
   = help: valid fragment specifiers are `ident`, `block`, `stmt`, `expr`, `pat`, `ty`, `lifetime`, `literal`, `path`, `meta`, `tt`, `item` and `vis`

error: aborting due to previous error

Couldn't compile the test.

failures:
    /tmp/mdbook-wjTpld/rust-2021/or-patterns-macro-rules.md - Or_patterns_in_macro_rules::Migration (line 54)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.03s


--- stderr

2021-11-03 05:43:15 [ERROR] (mdbook::utils): Error: One or more tests failed

Exited with code exit status 101
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,7 +3,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: quay.io/rust-lang-ja/circleci:edition-guide-mdbook-0.4.5
+      - image: quay.io/rust-lang-ja/circleci:edition-guide-mdbook-0.4.13
     steps:
       - checkout
       - run: rustc --version --verbose

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -24,9 +24,9 @@ jobs:
       - store_artifacts:
           path: /tmp/docs.txz
           destination: docs.txz
-      # - run:
-      #     name: Testing book
-      #     command: mdbook test
+      - run:
+          name: Testing book
+          command: mdbook test
       # - run:
       #     name: Check for broken links
       #     command: |

